h/static/scripts/discovery.coffee

```diff
@@ -86,6 +86,15 @@ module.exports = class Discovery
   _onMessage: (event) =>
     {source, origin, data} = event
 
+    # If `origin` is 'null' the source frame is a file URL or loaded over some
+    # other scheme for which the `origin` is undefined. In this case, the only
+    # way to ensure the message arrives is to use the wildcard origin. See:
+    #
+    #   https://developer.mozilla.org/en-US/docs/Web/API/Window/postMessage
```

Comment:
I couldn't see how to derive from [the spec](https://hyp.is/AVQ4OV7DH9ZO4OKSl1Cg/www.w3.org/TR/2008/WD-html5-20080610/comms.html) that "null" would definitely be used as the serialization of the origin in the case where the source frame is a file URL, or whether this value might be UA-defined.

I'm currently doing some testing in a few different browsers I have available.